frontend/wallet/src/main/Application.re

```diff
@@ -1,30 +1,207 @@
-module Action = {
-  type t =
-    | DoAction;
-};
+open Tc;
 
 module State = {
-  type t('a) = unit;
-};
+  module CodaProcessState = {
+    type t('proc) =
+      // the process is stopped
+      // it can be stopped unexpectedly with a string
+      | Stopped(Result.t(string, unit))
+      // we started it with the following extra arguments
+      | Started(list(string), Messages.CallTable.Ident.Encode.Set.t, 'proc)
+      // the process is ready for GraphQL interaction
+      | Ready(list(string), 'proc);
 
-let reduce = _acc =>
-  fun
-  | Action.DoAction => ();
+    let toString =
+      fun
+      | Stopped(res) => "Stopped(" ++ Js.String.make(res) ++ ")"
+      | Started(args, ident, proc) =>
+        "Started("
+        ++ Js.String.make(args)
+        ++ ";;"
+        ++ Js.String.make(ident)
+        ++ ";;"
+        ++ Js.String.make(proc)
+        ++ ")"
+      | Ready(args, proc) =>
+        "Ready("
+        ++ Js.String.make(args)
+        ++ ";;"
+        ++ Js.String.make(proc)
+        ++ ")";
```

Comment:
Printf.sprintf or `{j| |j}` if you need to leave this in? Unless we can just Js.log() the whole thing and get the right idea